### PR TITLE
allow setting basic auth headers for prometheus datasource

### PR DESCRIPTION
**Pull request by hagen1778**
#6799

### CODE REVIEW

**Comment by bergquist** [comment]
@hagen1778 the git commit auth have to match your github account. Otherwise you cannot sign the CLA.

File: pkg/tsdb/prometheus/prometheus.go [hunk 1]
```diff
@@ -22,6 +22,18 @@ type PrometheusExecutor struct {
...
+type basicAuthTransport struct {
+	*http.Transport
+
+	username string
+	password string
+}
+
+func (bat basicAuthTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	req.SetBasicAuth(bat.username, bat.password)
+	return http.DefaultTransport.RoundTrip(req)
```
Comment (bergquist) [inline pkg/tsdb/prometheus/prometheus.go]: I think the correct thing here would be to call `bat.Transport.Round()` rather then `http.DefaultTransport`.

**Review by bergquist — CHANGES_REQUESTED** [review]

File: pkg/tsdb/prometheus/prometheus.go (hunk 1 above)
Comment (hagen1778) [inline pkg/tsdb/prometheus/prometheus.go]: you are absolutely right

**Review by bergquist — APPROVED** [review]

**Comment by bergquist** [comment]
Great! Thank you for contributing! :)

**Pull request merged**